e2etest/go.mod

```diff
@@ -2,6 +2,11 @@ module go.universe.tf/e2etest
 
 go 1.21
 
+replace (
+	k8s.io/kubectl => k8s.io/kubectl v0.29.3
+	k8s.io/kubelet => k8s.io/kubelet v0.29.3
+)
+
```

Comment:
Something similar to this has occured last time: https://github.com/metallb/metallb/pull/2198#issuecomment-1865791977

Further information is,while I am using Jetbrain IDE developing, it runs `go list -m -json all` to obtain all dependencies at project loading, I guess. 

Without the replaces, if the project requires some kubernetes code whose version is set to v0.0.0,  the `go list` command would fail.

So this is something like a work around I think, but not sure if this is the best way.Also,  I am a little confused why this problem is not triggerd in github ci env

Reply:
yeah it's a bit odd, can you try removing it from the commit and see what happens in ci?

Reply:
> yeah it's a bit odd, can you try removing it from the commit and see what happens in ci?

Ci will work fine because this actuallly doesn't affect `go mod tidy`. Maybe I can put this in a temporary changelist to avoid IDE error.

Reply:
What if you go get those deps instead of replacing? Replacing pins the deps and it will require manual updates (plus, we made a considerable effort to avoid doing that), with `require` we can rely on dependabot.

Reply:
> What if you go get those deps instead of replacing? Replacing pins the deps and it will require manual updates (plus, we made a considerable effort to avoid doing that), with `require` we can rely on dependabot.

Got it. I tried `go get k8s.io/kubectl` and `go get k8s.io/kubelet` and this did solve the problem but meanwhile they were added to go.mod. Is it ok to submit this change to go.mod?

```
require(
...
	k8s.io/kubectl v0.29.4 // indirect
	k8s.io/kubelet v0.29.4 // indirect
...
)
```

Reply:
absolutely!

Reply:
@oribon I wonder if we should enforce `go list -m -json all` to work on CI, as we want to have a smooth dev experience (not only to those using vim :trollface: )

Reply:
> @oribon I wonder if we should enforce `go list -m -json all` to work on CI, as we want to have a smooth dev experience (not only to those using vim :trollface: )

Speaking of this, you can take a look at #2349 , which is a bug related to dev experience

Reply:
this has still to be replaced with require, right?

Reply:
> this has still to be replaced with require, right?

Seem right. Without replaces `go list` still reports errors
```
❯ cd e2etest
❯ go list -m -json all
go: k8s.io/kubectl@v0.0.0: invalid version: unknown revision v0.0.0
go: k8s.io/kubelet@v0.0.0: invalid version: unknown revision v0.0.0
```

Reply:
even with require, you mean?

Reply:
> even with require, you mean?

yes